### PR TITLE
Bundle MongoDB Atlas

### PR DESCRIPTION
**Pull request by kalafut**
This bundles two MongoDB Atlas plugins: one for database secrets, one for Programmatic API key secrets.

Docs are also included. The docs and api docs for the "database" section is complete. Docs for the programmatic secrets are included, but we're waiting on API docs.  You can see a recent preview here:

* https://5e3baa791a085b0009618ed5--vault-www.netlify.com/docs/secrets/databases/mongodbatlas/
* https://5e3baa791a085b0009618ed5--vault-www.netlify.com/api-docs/secret/databases/mongodbatlas/
* https://5e3baa791a085b0009618ed5--vault-www.netlify.com/docs/secrets/mongodbatlas/

### CODE REVIEW

**Comment by tyrannosaurus-becks** [comment]
Looks like some tests are failing related to this - the TestPredictPlugins ones are definitely related because they work off a separate list with the plugins hard-coded.

**Comment by kalafut** [comment]
doh. I updated one hard-coded list... I'll go find the rest.

File: website/pages/docs/secrets/mongodbatlas/index.mdx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    lease_renewable    true
+    description        vault-test-1563980947-1318
+    private_key        905ae89e-6ee8-40rd-ab12-613t8e3fe836
+    public_key         klpruxce
+```
+
+## API
+
+The MongoDB Atlas secrets engine has a full HTTP API. Please see the
+[MongoDB Atlas secrets engine API docs](/api/secret/mongodbatlas) for more details.
```
Comment (tyrannosaurus-becks) [inline website/pages/docs/secrets/mongodbatlas/index.mdx]: This link is broken. I'm not sure where it's intending to go, maybe here?
```
[MongoDB Atlas Database Plugin HTTP API](/api-docs/secret/databases/mongodbatlas)
```

File: website/pages/docs/secrets/databases/mongodbatlas.mdx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+## Usage
+
+After the secrets engine is configured and a user/machine has a Vault token with
+the proper permissions, it can generate credentials.
+
+1. Generate a new credential by reading from the `/creds` endpoint with the name
+of the role:
+
+    ```text
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/docs/secrets/databases/mongodbatlas.mdx]: Fixed

File: website/pages/api-docs/secret/databases/mongodbatlas.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+
+## Configure Connection
+
+In addition to the parameters defined by the [Database
+Backend](/api/secret/databases/index.html#configure-connection), this plugin
+has a number of parameters to further configure a connection.
+
+| Method   | Path                         |
+| :--------------------------- | :--------------------- |
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/api-docs/secret/databases/mongodbatlas.mdx]: Fixed

File: website/pages/api-docs/secret/databases/mongodbatlas.mdx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+### Sample Payload
+
+```json
+{
+  "plugin_name": "mongodbatlas-database-plugin",
+  "allowed_roles": "readonly",
+  "public_key": "aPublicKey",
+  "private_key": "aPrivateKey",
+  "project_id": "aProjectID"
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/api-docs/secret/databases/mongodbatlas.mdx]: sounds good

File: website/pages/docs/secrets/mongodbatlas/index.mdx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    documentation).
+
+Create a Vault role for a MongoDB Atlas Programmatic API Key by mapping appropriate arguments to the
+   organization or project designated:
+
+  - Organization API Key: Set `organization_id` argument with the appropriate
+    [Organization Level Roles](https://docs.atlas.mongodb.com/reference/user-roles/#organization-roles).
+  - Project API Key: Set `project_id` with the appropriate [Project Level Roles](https://docs.atlas.mongodb.com/reference/user-roles/#project-roles).
+  - Create a Organization Key and Assign to a project: creates an Organization
+    key and [Assigns]() it to a project, for this `project_id` and `organization_id`
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/docs/secrets/mongodbatlas/index.mdx]: The whole bullet is sort of messed up. I've ping the original author for some clarification.

File: website/pages/docs/secrets/mongodbatlas/index.mdx (hunk 1 above)
Comment (kalafut) [reply website/pages/docs/secrets/mongodbatlas/index.mdx]: The link is correct, but we don't have content yet.

File: website/pages/docs/secrets/mongodbatlas/index.mdx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    For more detailed instructions on how to create a Programmatic API Key in the Atlas UI, including
+    available roles, visit the [Programmatic API Key documenation](https://docs.atlas.mongodb.com/configure-api-access/#programmatic-api-keys).
+
+1. Once you have a MongoDB Atlas Programmatic Key pair, as created in the previous step, Vault can now
+   be configured to use it with MongoDB Atlas:
+
+    ```bash
+    $ vault write mongodbatlas/config \
+        public_key=yhltsvan \
+        private_key=2c130c23-e6b6-4da8-a93f-a8bf33218830
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/docs/secrets/mongodbatlas/index.mdx]: They're fake.

File: website/pages/docs/secrets/databases/mongodbatlas.mdx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    $ vault write database/config/my-mongodbatlas-database \
+        plugin_name=mongodbatlas-database-plugin \
+        allowed_roles="my-role" \
+        public_key="a-public-key" \
+        private_key="a-private-key" \
+        project_id="a-project-id"
+    ```
+
+1. Configure a role that maps a name in Vault to a MongoDB Atlas command that executes and
+   creates the Database User credential:
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/docs/secrets/databases/mongodbatlas.mdx]: Lower cased them

**Comment by kalafut** [comment]
@tyrannosaurus-becks I've updated almost everything. I'll slide in the API docs when they arrive. And I'd just removed that one really confusing bullet entirely. I have a TODO to add the API docs, so we can either leave this PR open or add a follow-up PR with that addition.

File: website/pages/api-docs/secret/databases/mongodbatlas.mdx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+- `creation_statements` `(string: <required>)` – Specifies the database
+  statements executed to create and configure a user. Must be a
+  serialized JSON object, or a base64-encoded serialized JSON object.
+  The object can optionally contain a "database_name", the name of
+  the authentication database to log into MongoDB. In Atlas deployments of
+  MongoDB, the authentication database is always the admin database. And
+  also must contain a "roles" array. This array contains objects that holds
+  a series of roles "roleName", an optional "databaseName" and "collectionName"
+  value. For more information regarding the `roles` field, refer to
+  [MongoDB Atlas documentation](https://docs.atlas.mongodb.com/reference/api/database-users-create-a-user/).
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/api-docs/secret/databases/mongodbatlas.mdx]: Their link does cover the general `roles` parameter an is probably ok.

File: website/pages/docs/secrets/mongodbatlas/index.mdx [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+sidebar_title: "MongoDB Atlas"
+description: |-
+  The MongoDB Atlas Secrets Engine for Vault generates MongoDB Atlas
+  Programmatic API Keys dynamically.
+---
+
+# MongoDB Atlas Secrets Engine
+
+The MongoDB Atlas Secrets Engine generates Programmatic API keys. This allows one to manage the
+lifecycle of these MongoDB Atlas secrets programmatically. The created MongoDB Atlas secrets are
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/docs/secrets/mongodbatlas/index.mdx]: I've removed it. It adds little to the content.

File: website/pages/docs/secrets/mongodbatlas/index.mdx [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+time-based and are automatically revoked when the Vault lease expires, unless renewed.
+
+This MongoDB Atlas Secrets Engine supports the creation of Programmatic API keys. Vault will create
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/docs/secrets/mongodbatlas/index.mdx]: Removed

File: website/pages/docs/secrets/mongodbatlas/index.mdx [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+
+## Setup
+
+Most Secrets Engines must be configured in advance before they can perform their functions. These
+steps are usually completed by an operator or configuration management tool.
+
+1. Enable the MongoDB Atlas Secrets Engine:
+
+    ```bash
+    $ vault secrets enable mongodbatlas
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/pages/docs/secrets/mongodbatlas/index.mdx]: When I remove it the coloring doesn't change. Also, it looks like our docs almost universally prefix the CLI commands with $, AFAICT.

[the remark this answers is not in the document]
Comment (tyrannosaurus-becks) [reply website/pages/docs/secrets/mongodbatlas/index.mdx]: Weird, not sure why that happened to me, might have been for a different kind of code excerpt.

**Pull request merged**